src/bittytax/conv/parsers/bitstamp.py

```diff
@@ -85,10 +85,104 @@ def parse_bitstamp(data_row: "DataRow", parser: DataParser, **_kwargs: Unpack[Pa
         raise UnexpectedTypeError(parser.in_header.index("Type"), "Type", row_dict["Type"])
 
 
+def parse_bitstamp_rfc4180(
+    data_row: "DataRow", parser: DataParser, **_kwargs: Unpack[ParserArgs]
+) -> None:
+    row_dict = data_row.row_dict
+    data_row.timestamp = DataParser.parse_timestamp(row_dict["Datetime"])
+
+    if row_dict["Type"] in ("Ripple deposit", "Deposit"):
+        data_row.t_record = TransactionOutRecord(
+            TrType.DEPOSIT,
+            data_row.timestamp,
+            buy_quantity=Decimal(row_dict["Amount"]),
+            buy_asset=row_dict["Amount currency"],
+            wallet=WALLET,
+        )
+    elif row_dict["Type"] in ("Ripple payment", "Withdrawal"):
+        data_row.t_record = TransactionOutRecord(
+            TrType.WITHDRAWAL,
+            data_row.timestamp,
+            sell_quantity=Decimal(row_dict["Amount"]),
+            sell_asset=row_dict["Amount currency"],
+            wallet=WALLET,
+        )
+    elif row_dict["Type"] == "Staking reward":
+        data_row.t_record = TransactionOutRecord(
+            TrType.STAKING,
+            data_row.timestamp,
+            buy_quantity=Decimal(row_dict["Amount"]),
+            buy_asset=row_dict["Amount currency"],
+            wallet=WALLET,
+        )
+
+    elif row_dict["Type"] == "Market":
+        if row_dict["Fee"]:
+            fee_quantity = Decimal(row_dict["Fee"])
+            fee_asset = row_dict["Fee currency"]
+        else:
+            fee_quantity = None
+            fee_asset = ""
+
+        if row_dict["Subtype"] == "Buy":
+            data_row.t_record = TransactionOutRecord(
+                TrType.TRADE,
+                data_row.timestamp,
+                buy_quantity=Decimal(row_dict["Amount"]),
+                buy_asset=row_dict["Amount currency"],
+                sell_quantity=Decimal(row_dict["Amount"]),
```

Comment:
This looks incorrect. I think it should be "Value" and "Value currency"?

Reply:
Yes, I believe you are correct, my mistake. Corrected and re-commited.